comp/api/api/apiimpl/internal/agent/agent_test.go

```diff
@@ -63,3 +67,435 @@ func TestSetupHandlers(t *testing.T) {
+			expectedError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var body []byte
+			var err error
+			if tt.payload != nil {
+				if str, ok := tt.payload.(string); ok {
+					body = []byte(str)
+				} else {
+					body, err = json.Marshal(tt.payload)
+					require.NoError(t, err)
+				}
+			}
+
+			req := httptest.NewRequest(tt.method, "/install-info/set", bytes.NewReader(body))
+			req.Header.Set("Content-Type", "application/json")
+			w := httptest.NewRecorder()
+
+			installinfo.HandleSetInstallInfo(w, req)
+
+			assert.Equal(t, tt.expectedStatus, w.Code)
+
+			if tt.expectedError {
+				var response installinfo.SetInstallInfoResponse
+				err = json.NewDecoder(w.Body).Decode(&response)
+				require.NoError(t, err)
+				assert.False(t, response.Success)
+				assert.NotEmpty(t, response.Message)
+			} else {
+				var response installinfo.SetInstallInfoResponse
+				err = json.NewDecoder(w.Body).Decode(&response)
+				require.NoError(t, err)
+				assert.True(t, response.Success)
+				assert.NotEmpty(t, response.Message)
+
+				reqGet := httptest.NewRequest("GET", "/install-info/get", nil)
+				installinfo.HandleGetInstallInfo(w, reqGet)
+				var info installinfo.InstallInfo
+				err = json.NewDecoder(w.Body).Decode(&info)
+				require.NoError(t, err)
+				assert.Equal(t, "ECS", info.Tool)
+				assert.Equal(t, "1.0", info.ToolVersion)
+				assert.Equal(t, "ecs-task-def-v1", info.InstallerVersion)
+			}
+		})
+	}
+}
+
+func TestHandleGetInstallInfo(t *testing.T) {
+	tests := []struct {
+		name           string
+		method         string
+		setupRuntime   bool
+		expectedStatus int
+	}{
+		{
+			name:           "valid GET request with runtime info",
+			method:         "GET",
+			setupRuntime:   true,
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "valid GET request without runtime info",
+			method:         "GET",
+			setupRuntime:   false,
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "invalid method",
+			method:         "POST",
+			setupRuntime:   false,
+			expectedStatus: http.StatusOK,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.setupRuntime {
+				info := &installinfo.InstallInfo{
+					Tool:             "ECS",
+					ToolVersion:      "1.0",
+					InstallerVersion: "ecs-task-def-v1",
+				}
+				body, err := json.Marshal(installinfo.SetInstallInfoRequest{
+					Tool:             info.Tool,
+					ToolVersion:      info.ToolVersion,
+					InstallerVersion: info.InstallerVersion,
+				})
+				require.NoError(t, err)
+				w := httptest.NewRecorder()
+				req := httptest.NewRequest("POST", "/install-info/set", bytes.NewReader(body))
+				req.Header.Set("Content-Type", "application/json")
+				installinfo.HandleSetInstallInfo(w, req)
+			}
+
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/install-info/get", nil)
+
+			installinfo.HandleGetInstallInfo(w, req)
```

Comment:
Same question as: `Should we use resp, err := ts.Client().Do(req) instead of calling the handler directly?`